src/Network/NWWebSocketResponse.cs

```diff
@@ -0,0 +1,86 @@
+//
+// NWWebSocketResponse.cs: Bindings the Network nw_browser_t API.
+//
+// Authors:
+//   Manuel de la Pena (mandel@microsoft.com)
+//
+// Copyrigh 2019 Microsoft Inc
+//
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using ObjCRuntime;
+using Foundation;
+using CoreFoundation;
+
+using OS_nw_ws_response=System.IntPtr;
+
+namespace Network {
+
+	[TV (13,0), Mac (10,15), iOS (13,0), Watch (6,0)]
+	public enum NWWebSocketResponseStatus {
+		Invalid = 0,
+		Accept = 1,
+		Reject = 2,
+	}
+
+	[TV (13,0), Mac (10,15), iOS (13,0), Watch (6,0)]
+	public class NWWebSocketResponse : NativeObject {
+
+		public NWWebSocketResponse (IntPtr handle, bool owns) : base (handle, owns) {}
+
+		[DllImport (Constants.NetworkLibrary, CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
+		static extern unsafe OS_nw_ws_response nw_ws_response_create (NWWebSocketResponseStatus status, string selected_subprotocol);
+
+		public NWWebSocketResponse (NWWebSocketResponseStatus status, string subprotocol)
+			=> InitializeHandle (nw_ws_response_create (status, subprotocol));
+
+		[DllImport (Constants.NetworkLibrary, CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
+		static extern unsafe string nw_ws_response_get_selected_subprotocol (OS_nw_ws_response response);
```

Comment:
```suggestion
		static extern string nw_ws_response_get_selected_subprotocol (OS_nw_ws_response response);
```

Reply:
Also I think you need to marshal the return value manually, because by default the managed runtime will try to free the native char* pointer.